packages/ui/src/models/visualization/flows/nodes/mappers/parallel-processor-base-node-mapper.test.ts

```diff
@@ -0,0 +1,123 @@
+  let mapper: MulticastNodeMapper | LoadBalanceNodeMapper;
+  let path: string;
+  let routeDefinition: unknown;
+
+  const testGetVizNodeFromProcessor = (processorName: string) => {
+    it(`should return a VisualizationNode for ${processorName}`, () => {
+      routeDefinition = {
+        from: {
+          uri: 'timer:timerName',
+          steps: [
+            {
+              [processorName]: {
+                id: `${processorName}-test`,
+              },
+            },
+          ],
+        },
+      };
+      const vizNode = mapper.getVizNodeFromProcessor(path, {} as ICamelElementLookupResult, routeDefinition);
+
+      expect(vizNode).toBeDefined();
+      expect(vizNode.data).toMatchObject({
+        path,
+        icon: expect.any(String),
+        processorName,
+        isGroup: true,
+      });
+    });
+
+    it(`should return a VisualizationNode with children for ${processorName}`, () => {
+      routeDefinition = {
+        from: {
+          uri: 'timer:timerName',
+          steps: [
+            {
+              [processorName]: {
+                id: `${processorName}-123`,
+                steps: [
+                  {
+                    log: {
+                      id: 'log-123',
+                      message: 'test',
+                    },
+                  },
+                  {
+                    log: {
+                      id: 'log-456',
+                      message: 'test',
+                    },
+                  },
+                ],
+              },
+            },
+          ],
+        },
+      };
+
+      const vizNode = mapper.getVizNodeFromProcessor(path, {} as ICamelElementLookupResult, routeDefinition);
+      expect(vizNode.getChildren()).toHaveLength(2);
+      expect(vizNode.getChildren()?.[0].getNextNode()).toBeUndefined();
+      expect(vizNode.getChildren()?.[1].getPreviousNode()).toBeUndefined();
+    });
+  };
+
+  // Common test logic for `addNodeInteraction`
+  const testAddNodeInteraction = () => {
+    it('should not allow child nodes to have previous/next steps', () => {
+      const ParentVizNode = createVisualizationNode('test', {});
+      const childVizNode = createVisualizationNode('child', {});
+      ParentVizNode.addChild(childVizNode);
+      mapper.addNodeInteraction(ParentVizNode, 'log');
+
+      expect(ParentVizNode.getChildren()).toHaveLength(1);
+      expect(ParentVizNode.getChildren()?.[0].getNodeInteraction().canHavePreviousStep).toEqual(false);
+      expect(ParentVizNode.getChildren()?.[0].getNodeInteraction().canHaveNextStep).toEqual(false);
+    });
+  };
+
+  describe('with multicast', () => {
+    beforeEach(() => {
+      const rootNodeMapper = new RootNodeMapper();
+      rootNodeMapper.registerDefaultMapper(mapper);
+
+      mapper = new MulticastNodeMapper(rootNodeMapper);
+
+      path = 'from.steps.0.multicast';
+    });
+
+    describe('getVizNodeFromProcessor', () => {
+      testGetVizNodeFromProcessor('multicast');

```

Comment:
There's another way to do this reusable testing: https://jestjs.io/docs/api#testeachtablename-fn-timeout

here's an example from [camel-on-exception-visual-entity.test.ts](https://github.com/KaotoIO/kaoto/blob/691534302c4686295e61c9f0e1d6f9d424817a63/packages/ui/src/models/visualization/flows/camel-on-exception-visual-entity.test.ts#L35-L53)
```typescript
  describe('getNodeInteraction', () => {
    it.each([
      'route',
      'from',
      'to',
      'log',
      'onException',
      'onCompletion',
      'intercept',
      'interceptFrom',
      'interceptSendToEndpoint',
    ])(`should return the correct interaction for the '%s' processor`, (processorName) => {
      const onExceptionDef = { onException: {} as OnException };
      const entity = new CamelOnExceptionVisualEntity(onExceptionDef);

      const result = entity.getNodeInteraction({ processorName });
      expect(result).toMatchSnapshot();
    });
  });
```

So in your case, you can create a simple string array with the processors you wanna check and then implement the test just once:

```typescript
it.each([
  'multicast',
  'loadBalance'
])(`should return a VisualizationNode for ${processorName}`, (processorName: string) => {
  // Existing test goes here
});
```

Same for the other test